site/server/app/com/fortysevendeg/exercises/persistence/repositories/UserProgressRepository.scala

```diff
@@ -0,0 +1,60 @@
+package com.fortysevendeg.exercises.persistence.repositories
+
+import com.fortysevendeg.exercises.persistence.PersistenceImpl
+import com.fortysevendeg.exercises.persistence.domain._
+import doobie.imports.ConnectionIO
+import shared.UserProgress
+
+import scalaz.Scalaz._
+
+trait UserProgressRepository {
+
+  def findByUserId(userId: Long): ConnectionIO[Option[UserProgress]]
+
+  def create(request: SaveUserProgress.Request): ConnectionIO[UserProgress]
+
+  def delete(id: Long): ConnectionIO[Boolean]
+
+  def update(userProgress: UserProgress): ConnectionIO[UserProgress]
+}
+
+class UserProgressDoobieRepository(implicit persistence: PersistenceImpl) extends UserProgressRepository {
+
```

Comment:
Since what it returns it's a Doobie type I don't think we need a facade but we do need to change the name so it's not Impl. It's not implementing a trait or contract per se but just a bunch of persistence related utils.